File: objectModel/Java/objectmodel/src/main/java/com/microsoft/commondatamodel/objectmodel/cdm/projections/CdmOperationCollection.java
// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT License. See License.txt in the project root for license information.

package com.microsoft.commondatamodel.objectmodel.cdm.projections;

import com.microsoft.commondatamodel.objectmodel.cdm.CdmCollection;
import com.microsoft.commondatamodel.objectmodel.cdm.CdmCorpusContext;
import com.microsoft.commondatamodel.objectmodel.cdm.CdmObject;
import com.microsoft.commondatamodel.objectmodel.enums.CdmObjectType;

/**
 * Collection of operations
 */
public class CdmOperationCollection extends CdmCollection<CdmOperationBase> {
    public CdmOperationCollection(final CdmCorpusContext ctx, final CdmObject owner) {
        super(ctx, owner, CdmObjectType.Error);
    }

    public CdmOperationBase add(CdmOperationBase operation) {
        return super.add(operation);
    }

    public boolean remove(CdmOperationBase operation) {
        return super.remove(operation);
    }
}

File: objectModel/Python/cdm/objectmodel/projections/cdm_operation_collection.py
# Copyright (c) Microsoft Corporation. All rights reserved.
# Licensed under the MIT License. See License.txt in the project root for license information.

from typing import TYPE_CHECKING

from cdm.enums import CdmObjectType
from cdm.objectmodel import CdmCollection

if TYPE_CHECKING:
    from cdm.objectmodel import CdmCorpusContext, CdmObject, CdmOperationBase


class CdmOperationCollection(CdmCollection):
    """Collection of operations"""

    def __init__(self, ctx: 'CdmCorpusContext', owner: 'CdmObject'):
        super().__init__(ctx, owner, CdmObjectType.ERROR)

    def append(self, operation: 'CdmOperationBase') -> 'CdmOperationBase':
        return super().append(operation)

    def remove(self, operation: 'CdmOperationBase') -> None:
        return super().remove(operation)
